libs9s/s9srpcclient.cpp

```diff
@@ -11849,6 +11860,93 @@ S9sRpcClient::syncClusters()
     return executeRequest(uri, request);
 }
 
+bool
+S9sRpcClient::registerRedisShardedCluster(
+        const S9sVariantList &hosts,
+        const S9sString      &osUserName,
+        const S9sString      &redisVersion)
+{
+    S9sOptions     *options = S9sOptions::instance();
+    S9sVariantMap   request;
+    S9sVariantMap   job = composeJob();
+    S9sVariantMap   jobData = composeJobData();
+    S9sVariantMap   jobSpec;
+    S9sString       uri = "/v2/jobs/";
+
+    if (hosts.size() < 1u)
+    {
+        PRINT_ERROR(
+                "Node is not specified while registering existing cluster.");
+        return false;
+    }
+    else if (hosts.size() > 1u)
+    {
+
+        PRINT_ERROR("Redis sharded clusters requires only one cluster's node to discover topology");
```

Comment:
If it requires only one node, have you considered to take the first one and proceed?

Reply:
I think it is more clear the command usaga this way